lectures/5-commitments.tex

```diff
@@ -0,0 +1,269 @@
+For example, if a set of balances is committed, the sum of any subset can be proven without revealing the exact value of each balance. 
+This is achieved by disclosing the sum of the balances and the corresponding sum of the blinding factors.
+
+\vspace{0.5 cm}
+
+\subsubsection{Vector commitments}
+
+Vector commitment schemes allows to commit to a vector of values rather than a value and a blinding term.
+
+\vspace{0.5 cm}
+
+\textbf{Pedersen Vector Commitments}
+
+Suppose we have a set of random elliptic curve points $(G_1, \ldots ,G_n)$ of cyclic group $\mathbb{G}$ 
+(that we do not know the discrete logarithm of), a vector $(m_1, m_2 \ldots m_n)$ and a random value $r$. We can do the following:
+
+\begin{center}
+    $ \mathcal{C} = m_1\cdot[G_1] + m_2 \cdot [G_2] \ldots + m_n \cdot [G_n] + r \cdot [Q]$ 
+\end{center}
+
+Since the \textit{Prover} does not know the discrete logarithm of the generators, they don`t know the discrete logarithm of [C]. 
+Hence, this scheme is binding: they can only reveal $(v_1, \ldots ,v_n)$ to produce $[C]$ later, they cannot produce another vector.
+
+Prover can later open the commitment by revealing the vector $(m_1, m_2 \ldots m_n)$ and a blinding term $r$.
+
+\vspace{0.5 cm}
+
+\textbf{Merkle Tree based Vector Commitments}
+
+A naive approach for a vector commitment would be hash the whole vector. More sophisticated scheme uses divide-and-conquer approach by building a binary tree out of vector elements.
+
+\begin{figure}[H]
+    \centering\includegraphics[width=0.9\linewidth, clip]{images/lecture_5/MerkleTree.png}
+
+    \caption{Merkle Tree structure}
+\end{figure}
+
+A Merkle Tree is a data structure to efficiently and securely verify the commiments to a vector of data. It is a binary tree where each leaf node represents a hash of a data block, and each non-leaf node is a hash of its child nodes' concatenated hashes. The top node, called the root hash or Merkle root, uniquely represents the entire data set. By comparing this root with a known valid root, one can quickly verify the authenticity and integrity of the data without needing to examine the entire dataset. 
+
+To prove the inclusion of element into the tree, a corresponding Merkle Branch is used. On the example below, $M_1$ inclusion is proved, and 
+$(M_2, H(M_3 \parallel M_4), H(H(M_5 \parallel M_6) \parallel H(M_7 \parallel M_8)))$ is an inclusion branch vector.
+
+\begin{figure}[H]
+    \centering\includegraphics[width=0.9\linewidth, clip]{images/lecture_5/MerkleTreeProof.png}
+
+    \caption{Merkle Tree inclusion proof branch}
+\end{figure}
+
+One of Merkle tree key advantages is that it allows for the selective disclosure of specific elements within the data set without revealing the rest. 
+
+\subsubsection{Polynomial commitment}
+
+Polynomial commitment can be used to prove that the commited polynomial satisfies certain properties (passes through a certain point $(x,y)$), without revealing what the polynomial is.
```

Comment:
This is the case for the univariate polynomial, but the polynomial commitments also work for multivariate (and typically linear) polynomials